### PR TITLE
fix: XYNE-61737 decode FlowJSON in automation trigger message

### PR DESCRIPTION
**Pull request by XyneSpaces**
Ticket: XYNE-61737

1. Problem Description:
When an automation is triggered by a FlowJSON alert message, the trigger passed the raw stored message content into the agent's task. FlowJSON is stored as `<div data-flow-json="...escaped JSON...">Flow JSON</div>` — the real alert text (metric, merchant, window) lives inside the `data-flow-json` attribute and the only visible text node is the literal string "Flow JSON". As a result agents (e.g. rcaAgent) received "Flow JSON" instead of the alert and had to call other tools (the spaces subagent) just to read what triggered them, making runs slower and noisier.

2. If this is a bug fix or an enhancement, how did you identify the issue?:
Traced the automation RUN_AGENT path. `run-agent.step.ts` passes the configured prompt through as the agent task and only adds conversationId + channelId as visible context. `hydrateMessageReceivedPayload` in `message-received.trigger.ts` set `message.content` straight from the raw stored `messageRow.content`, with no FlowJSON decoding — unlike notifications, Vespa indexing and mention scanning, which already decode FlowJSON.

3. Explain the solution implemented in the PR:
- Extracted the FlowJSON text helpers (`extractTextFromFlowJson`, `cleanNotificationText`, `extractCleanTextFromFlowJson`, `getFlowJsonContentForNotification`, `getFlowJsonRawTextForMentions`) into a new shared module `apps/backend/src/utils/flowJson.ts`. They were previously private to `zero/side-effects/tables/messages-handler.ts`, which itself imports the message-received trigger — so importing the decoder back into the trigger would have created an import cycle.
- `message-received.trigger.ts` now decodes content via `getFlowJsonContentForNotification(...) ?? rawContent`. The helper returns null for non-FlowJSON content, so normal messages pass through unchanged.
- `messages-handler.ts` now consumes the shared helpers and re-exports `getFlowJsonContentForNotification`, so the existing `zero/vespa-injection/core/mapper.ts` import is unchanged. No behavior change for notifications/indexing.

4. Documentation (link to docs created/updated for this change): N/A

5. DB Alter Details (if any): N/A

6. Data fix Details (if any): N/A

7. Environment variable Changes (if any): N/A

8. Testing (how it was tested; screenshots/links): `npx tsc --noEmit` passes clean on apps/backend. Manual validation: fire one RCA automation and confirm the agent run receives the decoded metric/merchant text (not the literal "Flow JSON") and no longer calls the spaces subagent for why_now. Note: card-only flows (e.g. plan/diff artifacts) whose text isn't in `content` props decode to empty and fall back to raw content, so those are never worse than before.

9. Services to which this change is to be deployed: xyne-spaces backend (automations + zero side-effects).

10. Main PR link (if this PR is raised against a release branch): N/A (raised against main)

### CODE REVIEW

**Review by XyneSpaces — COMMENTED** [review]
## Xyne Spaces Reviewer — PR #1401

Inline review finding posted for the changed hunk at head dce0fbd176301e6bd6e2cccf88881d9f2e0ebc26.

File: apps/backend/src/utils/flowJson.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ */
+
+/**
+ * Extract plaintext content strings from a FlowJSON payload for notification
+ * preview and mention scanning.
+ *
+ * Walks the component tree and collects every text `content` prop, then joins
+ * them. Returns '' for non-flow / unparseable content.
+ */
+export function extractTextFromFlowJson(content: string): string {
```
Comment (XyneSpaces) [inline apps/backend/src/utils/flowJson.ts:11]: **[major] R7 / Guideline 204** — This PR introduces a new FlowJSON parser/cleaner that now drives automation trigger payloads, but no unit test was added for the new behavior or edge cases (normal HTML fallback, escaped `data-flow-json`, nested `children`, malformed JSON, and empty card-only content). Please add focused tests for `getFlowJsonContentForNotification`/`extractTextFromFlowJson` so future changes don't silently regress agent inputs.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1401

**Verdict:** Changes requested
**Reviewed:** 3 files, 3 hunks · **Findings:** 0 blocking, 1 major, 0 nits
**Head SHA:** `dce0fbd176301e6bd6e2cccf88881d9f2e0ebc26`

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| major | R7 / Guideline 204 | `apps/backend/src/utils/flowJson.ts:24` | New FlowJSON parsing/cleaning behavior now drives automation trigger payloads, but the PR adds no unit tests for escaped FlowJSON, nested children, malformed JSON, non-FlowJSON fallback, or empty card-only payloads. |

### Recommended reviewer
**@uayush01** (recommended; review request not submitted because the available GitHub toolset has no request-reviewers endpoint) — backend/Zero-active candidate with low explicit review-request load.

| Candidate | Owns (files in this PR) | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-------------------------|-------------------|-----------------|-------------|
| @uayush01 | backend/Zero-active recent author proxy; exact path ownership unavailable | no CODEOWNERS | 0 explicit requests found | Spaces user resolved; WFH signal |
| @MukherjeeSourish | backend automation/Zero-active recent author proxy; exact path ownership unavailable | no CODEOWNERS | could not compute reliably | active today in Spaces |
| @monika-annamaneni | backend-active recent author proxy; exact path ownership unavailable | no CODEOWNERS | 0 explicit requests found | active today |
| @Dnyaneshwari-04 | backend-active recent author proxy; exact path ownership unavailable | no CODEOWNERS | 0 explicit requests found | active today in #xyne-spaces |
| @Mayankbansal-juspay | backend-active recent author proxy; exact path ownership unavailable | no CODEOWNERS | not computed | active today |

No `.github/CODEOWNERS` file exists, and the PR had no requested reviewers. I used recent main-branch authors as the fallback ownership signal and Spaces activity to break ties. Please request **@uayush01** manually if the review request is still absent.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1401

**Verdict:** Changes requested
**Reviewed:** 3 files, 3 hunks · **Findings:** 0 blocking, 1 major, 0 nits

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| major | R7 / Guideline 204 | apps/backend/src/utils/flowJson.ts:11 | New FlowJSON parser/cleaner behavior is used by automation trigger payloads, but there is no focused unit coverage for decoded FlowJSON, normal fallback, malformed JSON, nested children, and empty card-only content. Existing inline comment already posted; not duplicated on this re-run. |

### Recommended reviewer
**@gyash1512** (review request not sent — this runtime has no GitHub request-reviewers tool/gh CLI available) — has same-file review familiarity and a confidently mapped Spaces user, but high current review load; `mahekag19` has the best workload/familiarity signal but could not be confidently mapped to a Spaces email.

| Candidate | Owns (files in this PR) | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-------------------------|-------------------|-----------------|-------------|
| @mahekag19 | messages-handler.ts + prior FlowJSON ingestion review | no CODEOWNERS file | 31 | best workload/familiarity, Spaces email not confirmed |
| @gyash1512 | prior messages-handler.ts review | no CODEOWNERS file | 274 | confidently maps to `yash.gupta@juspay.in` |
| @knaman2609 | prior messages-handler.ts review | no CODEOWNERS file | 278 | confidently maps to `naman.kalkhuria@juspay.in` |

Rationale: repo has no CODEOWNERS and no current requested reviewers. Because `mahekag19` could not be confidently mapped to a Spaces email, I recommend the strongest confidently taggable reviewer, @gyash1512, despite the larger pending-review queue. GitHub reviewer request could not be sent from this runtime because the available GitHub tools do not expose request-reviewers and no `gh` CLI tool is available.

**Comment by XyneSpaces** [comment]
## Xyne Spaces Reviewer — PR #1401

**Verdict:** Changes requested
**Reviewed:** 3 files, 3 hunks · **Findings:** 0 blocking, 1 major, 0 nits

### Findings
| Severity | Rule | Location | Issue |
|----------|------|----------|-------|
| major | R7 / Guideline 204 | `apps/backend/src/utils/flowJson.ts:11` | New FlowJSON parsing/cleaning behavior now feeds automation trigger payloads, but no focused unit tests were added for escaped FlowJSON, nested children, malformed JSON, empty/card-only content, and plain-message fallback. Existing inline comment already covers this; not reposted. |

### Recommended reviewer
**@sakthin-juspay** (recommended; review request not assigned because the available GitHub toolset has no request-reviewer endpoint/CLI fallback) — mapped to a real Spaces user and has the lowest pending review load among mapped candidates.

| Candidate | Owns (files in this PR) | Default reviewer? | Pending reviews | Tie-breaker |
|-----------|-------------------------|-------------------|-----------------|-------------|
| @sakthin-juspay | unavailable (commits-by-path API not exposed) | no CODEOWNERS file | 1 | mapped to Sakthi Narayanan R / `sakthi.n@juspay.in` |
| @PiyushJuspay | unavailable (commits-by-path API not exposed) | no CODEOWNERS file | 2 | mapped to Piyush Kesharwani / `piyush.k.001@juspay.in` |
| @utkarshpandey6 | unavailable | no CODEOWNERS file | 8 | Spaces mapping unavailable |
| @nikunjJuspay | unavailable | no CODEOWNERS file | 8 | Spaces mapping unavailable |

Rationale: `.github/CODEOWNERS` is absent, commits-by-path ownership was unavailable through the exposed GitHub tools, so the pick falls back to the mapped candidate with the lowest current review load: @sakthin-juspay.

Top blocking issue: none. Comments posted by this run: 1 summary, 0 new inline (existing R7 inline comment was not duplicated).

**Review by anuragdvd — APPROVED** [review]

**Review by anuragdvd — DISMISSED** [review]

**Review by satvikbatra — APPROVED** [review]

**Review by Mohan0544 — APPROVED** [review]
